go.mod

```diff
@@ -36,33 +45,58 @@ require (
+	github.com/prometheus/client_model v0.6.1 // indirect
+	github.com/prometheus/common v0.55.0 // indirect
+	github.com/prometheus/procfs v0.15.1 // indirect
 	github.com/shopspring/decimal v1.4.0 // indirect
+	github.com/sirupsen/logrus v1.9.3 // indirect
 	github.com/spf13/cast v1.6.0 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
-	github.com/stolostron/kubernetes-dependency-watches v0.9.1 // indirect
-	golang.org/x/crypto v0.23.0 // indirect
+	github.com/stolostron/go-log-utils v0.1.2 // indirect
+	github.com/stolostron/kubernetes-dependency-watches v0.10.0 // indirect
+	github.com/x448/float16 v0.8.4 // indirect
+	github.com/xlab/treeprint v1.2.0 // indirect
+	go.starlark.net v0.0.0-20230525235612-a134d8f9ddca // indirect
+	go.uber.org/multierr v1.11.0 // indirect
+	go.uber.org/zap v1.27.0 // indirect
+	golang.org/x/crypto v0.24.0 // indirect
 	golang.org/x/exp v0.0.0-20240525044651-4c93da0ed11d // indirect
-	golang.org/x/net v0.25.0 // indirect
-	golang.org/x/oauth2 v0.20.0 // indirect
-	golang.org/x/sys v0.20.0 // indirect
-	golang.org/x/term v0.20.0 // indirect
-	golang.org/x/text v0.15.0 // indirect
+	golang.org/x/mod v0.17.0 // indirect
+	golang.org/x/net v0.26.0 // indirect
+	golang.org/x/oauth2 v0.21.0 // indirect
+	golang.org/x/sync v0.7.0 // indirect
+	golang.org/x/sys v0.21.0 // indirect
+	golang.org/x/term v0.21.0 // indirect
+	golang.org/x/text v0.16.0 // indirect
 	golang.org/x/time v0.5.0 // indirect
-	google.golang.org/protobuf v1.34.1 // indirect
+	gomodules.xyz/jsonpatch/v2 v2.4.0 // indirect
+	google.golang.org/protobuf v1.34.2 // indirect
+	gopkg.in/evanphx/json-patch.v4 v4.12.0 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
-	k8s.io/api v0.29.5 // indirect
-	k8s.io/apimachinery v0.29.5 // indirect
-	k8s.io/client-go v0.29.5 // indirect
+	k8s.io/api v0.31.1 // indirect
+	k8s.io/apiextensions-apiserver v0.31.0 // indirect
+	k8s.io/apimachinery v0.31.1 // indirect
+	k8s.io/cli-runtime v0.28.10 // indirect
+	k8s.io/client-go v0.31.0 // indirect
 	k8s.io/klog v1.0.0 // indirect
-	k8s.io/klog/v2 v2.120.1 // indirect
+	k8s.io/klog/v2 v2.130.1 // indirect
 	k8s.io/kube-openapi v0.0.0-20240521193020-835d969ad83a // indirect
-	k8s.io/utils v0.0.0-20240502163921-fe8a2dddb1d0 // indirect
-	sigs.k8s.io/controller-runtime v0.17.5 // indirect
+	k8s.io/kubectl v0.28.10 // indirect
+	k8s.io/utils v0.0.0-20240711033017-18e509b52bc8 // indirect
+	sigs.k8s.io/controller-runtime v0.19.0 // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
+	sigs.k8s.io/kustomize/api v0.17.2 // indirect
+	sigs.k8s.io/kustomize/kyaml v0.17.1 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
+
+replace open-cluster-management.io/config-policy-controller => github.com/stolostron/config-policy-controller v0.0.0-20240926134553-da95cec15564
```

Comment:
I think this is going to break `go install`, which isn't ideal. I wonder: What would be the effect of changing the `go.mod` in `config-policy-controller` (and the propagator, while we're at it)? It perhaps wouldn't be great for magic-mirror, but it'd let us do away with all these `replace` statements. The alternative would be to use the equivalent upstream commit:
```
go get open-cluster-management.io/config-policy-controller@main
```

Reply:
Dependabot doesn't handle these sorts of pseudo versions, but it'd be easy to have a GitHub Action that does.

Reply:
> What would be the effect of changing the go.mod in config-policy-controller

What do you mean, what part are you thinking of changing?

I think if we changed the module name in stolostron/config-policy-controller, we would have to change it throughout its code.

Reply:
Yes, that's what I mean. It'd let us do `go get github.com/stolstron/config-policy-controller`. But really, I think we should use the upstream commits in `main` so that `go install` works and then if we need to do a patch for a previous ACM version, we can use a `replace` statement here. What would you think of that?

Reply:
The more I think about it, the more I think we should reference the upstream across our repos except in cases where it needs to be patched by the release branch. It'd make it cleaner and more maintainable overall...

Reply:
I thought there was some productization reason we needed to use the stolostron code in our releases. But since I can't find the policy, I (or someone else) might have made that up...

Reply:
We use the upstream for server-foundation, so I'm not sure it'd be any different. I think we've just been doing it and not considering any other way. I've added it to the retro to discuss it. 🙂

Reply:
All of these repos are replicated to DistGit for downstream, so maybe that's how the policy is fulfilled?

Reply:
...or not. Looks like DistGit is only container repos.